src/mercury-agent/mercury/agent/configuration.py

```diff
@@ -16,29 +16,31 @@
 import logging
 
 from mercury.common.configuration import get_configuration
-from mercury.common.clients.rpc.backend import BackEndClient
+from mercury.common.exceptions import MercuryConfigurationError
 
 __all__ = ['AGENT_CONFIG_FILE', 'agent_configuration']
 
-# Private
-__backend_client = None
 
 log = logging.getLogger(__name__)
 
 # Helpers
-AGENT_CONFIG_FILE = 'mercury-agent.yaml'
-agent_configuration = get_configuration(AGENT_CONFIG_FILE)
-remote_configuration = agent_configuration.get('remote', {})
-
-try:
-    backend_url = remote_configuration['rpc_service']
-except KeyError:
-    log.warning('Missing inventory service url')
-
-
-def get_backend_client():
-    # TODO: Trying this out, 0mq says it is ok
-    global __backend_client
-    if not __backend_client:
-        __backend_client = BackEndClient(backend_url)
-    return __backend_client
+DEFAULT_AGENT_CONFIG_FILE = 'mercury_agent.yaml'
+
+
+agent_configuration = {}
+remote_configuration = {}
+
+
+def set_agent_configuration(namespace):
+    global agent_configuration
+    global remote_configuration
```

Comment:
IIRC Don't need to specify globals for objects you are mutating, only for reassignment.